public/reorderSiteAwards.php

```diff
@@ -7,97 +7,275 @@
     abort(401);
 }
 
+function getInitialSectionOrders(array $gameAwards, array $eventAwards, array $siteAwards): array
+{
+    $awardsArrays = [
+        'gameAwards' => $gameAwards,
+        'eventAwards' => $eventAwards,
+        'siteAwards' => $siteAwards,
+    ];
+
+    $firstDisplayOrders = [];
+
+    foreach ($awardsArrays as $key => $awardsArray) {
+        $firstDisplayOrder = null;
+
+        foreach ($awardsArray as $award) {
+            if ($award['DisplayOrder'] >= 0) {
+                $firstDisplayOrder = $award['DisplayOrder'];
+                break;
+            }
+        }
+
+        $firstDisplayOrders[$key] = $firstDisplayOrder;
+    }
+
+    asort($firstDisplayOrders); // Sort the array while maintaining the key association
+
+    // Assign unique order values
+    $order = 1;
+    foreach ($firstDisplayOrders as $key => $value) {
+        if ($value !== null) {
+            $firstDisplayOrders[$key] = $order++;
+        }
+    }
+
+    // Replace null values with unique incrementing values, unless the array is empty
+    foreach ($firstDisplayOrders as $key => $value) {
+        if ($value === null && count($awardsArrays[$key]) > 0) {
+            $firstDisplayOrders[$key] = $order++;
+        }
+    }
+
+    return [
+        $firstDisplayOrders['gameAwards'],
+        $firstDisplayOrders['eventAwards'],
+        $firstDisplayOrders['siteAwards'],
+    ];
+}
+
+function generateManualMoveButtons(
+    int $awardCounter,
+    int $upValue,
+    int $downValue,
+    int $arrowCount = 1,
+    bool $autoScroll = false,
+    string $orientation = 'vertical',
+): string {
+    $arrowUp = '↑';
+    $arrowDown = '↓';
+
+    if ($arrowCount === 2) {
+        $arrowUp = '↑↑';
+        $arrowDown = '↓↓';
+    } elseif ($arrowCount === 3) {
+        $arrowUp = '↑↑↑';
+        $arrowDown = '↓↓↓';
```

Comment:
These should have tooltips. It's unclear what the difference is between them. Maybe it would be better to just use appropriate text labels (i.e. `↑10` or `↑Top`)

Reply:
Also, wouldn't it be easier to just pass the label in? `arrowCount` doesn't seem to be used for anything else.

Reply:
I have tested several different treatments for these button labels. I am interested in which you think works best:

A
![Screenshot 2023-03-24 at 8 13 52 PM](https://user-images.githubusercontent.com/3984985/227670352-aca0ad60-a5d0-44d4-a01f-d366050c1964.png)

B
![Screenshot 2023-03-24 at 8 15 03 PM](https://user-images.githubusercontent.com/3984985/227670405-aaffb1a2-4af3-407f-8e47-4c3e92435425.png)

C
![Screenshot 2023-03-24 at 8 17 20 PM](https://user-images.githubusercontent.com/3984985/227670451-3b4bb61c-b314-4bdc-ab95-45d5934d78ed.png)

Reply:
I like option C, though I'm struggling with the "up/down 50" items. The first time I clicked on the down 50, I couldn't find where the item went because it had moved down like four screens (and I was looking for it down two and a half screens or so). If we're going to keep that functionality, I'd recommend scrolling to the new location, or including numbers on the row so the user has an idea of what N+50 is.